<a href="https://colab.research.google.com/github/c-susan/datasci_5_statistics/blob/main/python_statistics.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Loading in Packages

In [25]:
import pandas as pd
from scipy.stats import chi2_contingency as chi2, ttest_ind

## **1. Chi-Square Test**

In [26]:
df1 = pd.read_csv('Order_and_Referring_10_3_2023.csv')
df1

NPI      LAST_NAME FIRST_NAME PARTB DME HHA PMD
0        1558467555       .MCINDOE     THOMAS     Y   Y   Y   Y
1        1417051921        A BELLE          N     Y   Y   Y   Y
2        1972040137      A NOVOTNY  ELIZABETH     Y   Y   Y   Y
3        1760465553       A SATTAR   MUHAMMAD     Y   Y   Y   Y
4        1295400745         A'NEAL     BROGAN     Y   Y   N   N
...             ...            ...        ...   ...  ..  ..  ..
1785834  1336502301           ZYZO       JOHN     Y   Y   Y   N
1785835  1225502768          ZZIWA     JACKIE     N   Y   N   Y
1785836  1124277249  ZZIWA-KABENGE      IRYNE     Y   Y   Y   Y
1785837  1033160296    ZZIWAMBAZZA     NATHAN     Y   Y   Y   Y
1785838  1326140070          TOUPS    JEFFREY     Y   N   N   N

[1785839 rows x 7 columns]

In [27]:
## View the values counts of selected columns
# 'DME' = Indicates whether provider can order Durable Medical Equipment
df1['DME'].value_counts()

Y    1785838
N          1
Name: DME, dtype: int64

In [28]:
# 'PMD' = Indicates whether provider can order Power Mobility Devices
df1['PMD'].value_counts()

Y    1475449
N     310390
Name: PMD, dtype: int64

In [29]:
## Creating a contingency table selected columns
contingency_table = pd.crosstab(df1['DME'], df1['PMD'])
contingency_table

PMD       N        Y
DME                 
N         1        0
Y    310389  1475449

In [30]:
## Using the contingency table to perform a chi-square test
chi2, p, degf, expected = chi2(contingency_table)
print(f"Chi2 value: {chi2}")
print(f"P-value: {p}")
print(f'Degrees of freedom: {degf}')
print(f'Expected Frequency: {expected}')

Chi2 value: 0.7409760431083185
P-value: 0.3893483960070897
Degrees of freedom: 1
Expected Frequency: [[1.73806261e-01 8.26193739e-01]
 [3.10389826e+05 1.47544817e+06]]


### **Summary**
**Question:** Is there an association between whether a provider can order Durable Medical Equipment (DME) and whether they can order Power Mobility Devices (PMD)?

**H0:** There is no relationship between DME and PMD (independent).

**H1:** There is a relationship between DME and PMD (dependent).

**Interpretation**

Since the p-value of 0.389 is more than the significance level of 0.05, we do not reject the null hypothesis. There is no relationship between a provider's ability to order DME and their ability to order PMD (DME is independent of PMD).

**______________________________________________________________________________________________________________**

## **2. T-Test**

**______________________________________________________________________________________________________________**

## **3. ANOVA**

**______________________________________________________________________________________________________________**

## **4. Regression Analysis**